Sources/BraintreeCard/BTCardClient.swift

```diff
@@ -0,0 +1,218 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// Used to process cards
+@objc public class BTCardClient: NSObject {
+
+    // MARK: - Internal Properties
+
+    /// Exposed for testing to get the instance of BTAPIClient
+    let apiClient: BTAPIClient
+
+    let graphQLTokenizeFeature: String = "tokenize_credit_cards"
+
+    // MARK: - Initializer
+
+    /// Creates a card client
+    /// - Parameter apiClient: An API client
+    @objc(initWithAPIClient:)
+    public init(apiClient: BTAPIClient) {
+        self.apiClient = apiClient
+    }
+
+    // MARK: - Public Methods
+
+    /// Tokenizes a card
+    /// - Parameters:
+    ///    - card: The card to tokenize.
+    ///    - completion: A completion block that is invoked when card tokenization has completed. If tokenization succeeds,
+    ///    `tokenize` will contain a nonce and `error` will be `nil`; if it fails, `tokenize` will be `nil` and `error`will describe the failure.
+    @objc(tokenizeCard:completion:)
+    public func tokenize(_ card: BTCard, completion: @escaping (BTCardNonce?, Error?) -> Void) {
+        let request = BTCardRequest(card: card)
+
+        apiClient.fetchOrReturnRemoteConfiguration() { configuration, error in
+            guard let configuration, error == nil else {
+                completion(nil, error)
+                return
+            }
+
+            if self.isGraphQLEnabled(for: configuration) {
+                if request.card.authenticationInsightRequested && request.card.merchantAccountID == nil {
+                    completion(nil, BTCardError.integration)
+                    return
+                }
+
+                let parameters = request.card.graphQLParameters()
+
+                self.apiClient.post("", parameters: parameters, httpType: .graphQLAPI) { body, _, error in
+                    if let error = error as NSError? {
+                        if error.code == BTCoreConstants.networkConnectionLostCode {
+                            self.apiClient.sendAnalyticsEvent("ios.tokenize-card.graphQL.network-connection.failure")
+                        }
+
+                        let response: HTTPURLResponse? = error.userInfo[BTCoreConstants.urlResponseKey] as? HTTPURLResponse
+                        var callbackError: Error? = error
+
+                        if response?.statusCode == 422 {
+                            callbackError = self.constructCallbackError(with: error.userInfo, error: error)
```

Comment:
can the error.userInfo be nil?

Reply:
Good catch! Yes, there could potentially be a situation where when we cast `error` to `NSError` above and the error is `nil` it'd still be cast successfully (while unlikely, certainly possible). Since a 422 would indicate an error from the gateway, I think we have always just assumed the gateway would return an error containing `userInfo`. I think we can just make the dictionary optional in the method to match what we were doing in Obj-C. Updated here: https://github.com/braintree/braintree_ios/pull/983/commits/d623ffc077af4bb9171937021efe95cf08972690